### DISCUSSION TITLE
removing nodes in the graph with degree 1

### DISCUSSION THREAD

**Discussion by chinmay5** [pyg-team/pytorch_geometric#2764] [Q&A]
I am parsing a knowledge graph and trying to build a small subgraph from it. After parsing I realized I have a lot of nodes with degree 1. I would ideally like to remove such nodes since they do not add much information. Is there a way of removing nodes with degree 1?

I know networkx has a function that can remove the nodes based on conditions. Hence, I convert my data into networkx and do the processing then convert it back to pytorch_geometric. I was wondering if there exists a simpler way for the same.

**Comment by rusty1s** [comment]
```python
from torch_geometric.utils import degree, subgraph

deg = degree(data.edge_index[1], data.num_nodes)
mask = deg == 1

edge_index, edge_attr = subgraph(mask, data.edge_index, data.edge_attr, relabel_nodes=True, num_nodes=data.num_nodes)

data.x = data.x[mask]
data.edge_index = edge_index
data.edge_attr = edge_attr
```

**Reply by chinmay5** [reply]
Shouldn't the mask be
`mask = deg >=2` so that we get hold of nodes that have a degree greater than 2 in the subgraph?

**Reply by rusty1s** [reply]
Yes, you are correct. Sorry for the confusion.
